.github/workflows/pipeline.yml

```diff
@@ -1,55 +1,49 @@
 name: CI-Checks
-on: 
-  push:
-    branches-ignore:
-      -master
-      -release-*
-jobs:
-  check-newsfile:
+on: [push]
+jobs: 
+  check-code-style:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
           python-version: '3.7'
-      - run: python -m pip install towncrier
-      - run: python -m towncrier.check
-    # plugins:
-    #   - docker#v3.0.1:
-    #       image: "python:3.7"
-    #       mount-buildkite-agent: false
-    # branches: "!master !release-*"
-
-  # - command:
-  #     - "python -m pip install tox"
-  #     - "tox -e check_codestyle"
-  #   label: "Check Code Style"
-  #   plugins:
-  #     - docker#v3.0.1:
-  #         image: "python:3.7"
-  #         mount-buildkite-agent: false
-
-  # - command:
-  #     - "python -m pip install tox"
-  #     - "tox -e check_types"
-  #   label: "Check Types (:mypy:)"
-  #   plugins:
-  #     - docker#v3.0.1:
-  #         image: "python:3.7"
-  #         mount-buildkite-agent: false
-
-  # - wait
-
-  # - command:
-  #     - "python -m pip install -e ."
-  #     - "trial tests"
-  #   label: "Run unit tests (on SQLite)"
-  #   plugins:
-  #     - docker#v3.0.1:
-  #         image: "python:3.7"
-  #         mount-buildkite-agent: false
+      - run: python -m pip install tox
+      - run: tox -e check_codestyle
+  
+  check-types-mypy:
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - run: python -m pip install tox  
+      - run: tox -e check_types
 
-  # - command:
+  run-unit-tests-sqlite:
+    needs: [check-code-style, check-types-mypy]
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - run: python -m pip install -e .
+      - run: trial tests
+  
+  # run-unit-tests-(on :postgres:):
+  #   needs: [run-unit-tests-sqlite]
```

Comment:
Having the postgres test `need`ing the sqlite test will cause them to run sequentially, whereas ideally they would run in parallel. I'm not sure if there's a better way to structure this other than adding `needs: [check-code-style, check-types-mypy]` to each job.

Reply:
> I'm not sure if there's a better way to structure this other than adding needs: `[check-code-style, check-types-mypy]` to each job.

There isn't, unfortunately. In Synapse I made a dummy task called `linting-done` like:

```yaml
  # Dummy step to gate other tests on without repeating the whole list
  linting-done:
    needs: [...]
    runs-on: ubuntu-latest
    steps:
      - run: "true"
```

And then all the subsequent jobs could just `needs: linting-done`

But that's almost certainly overkill for Sygnal's test matrix